assets/js/state/lastExecutions.js

```diff
@@ -42,11 +44,36 @@ export const lastExecutionsSlice = createSlice({
 
       const lastExecutionState = {
         ...initialExecutionState,
+        executionStarted: true,
         error,
       };
 
       state[groupID] = lastExecutionState;
     },
+    setExecutionStarted: (state, { payload }) => {
+      const { groupID: clusterID, checks: startedChecks } = payload;
+
+      // The execution started event contains the information
+      // about the checks "really" started in the execution
+      // this checks could differ from the ones requested
+      // because for some reason the checks could be skipped by
+      // some conditions
+      // we want to make sure that only the real checks executed
+      // are present into the state when an execution occur
+
+      // eslint-disable-next-line prefer-destructuring
+      const targets = state[clusterID].data.targets;
+      const targetsWithCheckUsedInExecution = targets.map((target) => ({
+        ...target,
+        checks: target.checks.filter((checkID) =>
+          startedChecks.includes(checkID)
+        ),
+      }));
+
+      state[clusterID].data.targets = targetsWithCheckUsedInExecution;
```

Comment:
This would better be:
```
const lastExecutionState = {
        ...initialExecutionState,
        executionStarted: true,
        data: {
          status: 'running',
          targets,
        },
      }
```
Or similar, instead of dot access assignment

Reply:
I think dot assignment is fully justified here, it's one of the reason we use redux toolkit because we don't have to deal with immutability and reference

Reply:
+1 to keep the dot notation as Redux toolkit uses [Immer](https://github.com/immerjs/immer) internally

Reply:
It was more about aesthetics, I find it simply more beatiful/readable. But not hard opinion